source/client/process_impl.cc

```diff
@@ -100,123 +100,154 @@ class BootstrapFactory : public Envoy::Logger::Loggable<Envoy::Logger::Id::main>
   }
 };
 
-// Implementation of Envoy::Server::Instance used as a placeholder. None of its methods
-// should be called because Nighthawk is not a real Envoy that performs xDS config validation.
-class NullServerInstance : public Envoy::Server::Instance {
-  Envoy::Server::Admin& admin() override { PANIC("not implemented"); }
-  Envoy::Api::Api& api() override { PANIC("not implemented"); }
-  Envoy::Upstream::ClusterManager& clusterManager() override { PANIC("not implemented"); }
+// Implementation of Envoy::Server::Instance. Only methods used by Envoy's code
+// when Nighthawk is running are implemented.
+class NighthawkServerInstance : public Envoy::Server::Instance {
```

Comment:
I did that yesterday and after digging deeper I believe that this is still our best path forward. Running the actual instance of Envoy server in Nighthawk client causes complications with lifetime, since the Server had non-trivial startup and shutdown procedure that we don't control. Additionally using the real instance may also change Nighthawk's functionality.

Unless we have a good reason to use the real objects, we probably don't want to invest into it.